### PR TITLE
wgengine/wgcfg: always close io.Pipe

### PR DESCRIPTION
**Pull request by josharian**
In DeviceConfig, we did not close r after calling FromUAPI.
If FromUAPI returned early due to an error, then it might
not have read all the data that IpcGetOperation wanted to write.
As a result, IpcGetOperation could hang, as in #3220.

We were also closing the wrong end of the pipe after IpcSetOperation
in ReconfigDevice.

To ensure that we get all available information to diagnose
such a situation, include all errors anytime something goes wrong.

This should fix the immediate crashing problem in #3220.
We'll then need to figure out why IpcGetOperation was failing.

Signed-off-by: Josh Bleecher Snyder <josh@tailscale.com>

### CODE REVIEW

**Comment by josharian** [comment]
Tests pass again. :) PTAL

File: wgengine/magicsock/magicsock_test.go [hunk 1]
```diff
@@ -296,10 +297,19 @@ func meshStacks(logf logger.Logf, mutateNetmap func(idx int, nm *netmap.NetworkM
...
 			if err := m.Reconfig(wg); err != nil {
-				if ctx.Err() != nil || errors.Is(err, errConnClosed) {
+				if ctx.Err() != nil {
 					// shutdown race, don't care.
 					return
 				}
+				var me multierror.MultipleErrors
+				if errors.As(err, &me) {
+					for _, e := range me {
+						if errors.Is(e, errConnClosed) {
```
Comment (crawshaw) [inline wgengine/magicsock/magicsock_test.go:307]: woah, does an errors.Is not see into a multierror? that seems wrong. does unwrap make it impossible to implement an iterator?

**Review by crawshaw — APPROVED** [review]

**Comment by crawshaw** [comment]
oh it totally can, `errors.Is` looks for a method named `Is`, so a multierror can implement that and do the iteration.

looks like multierror hasn't been touched in 6 years. want to fork it under github.com/tailscale/multierror and implement Is, instead of putting the loop in the test?

**Comment by josharian** [comment]
Now with a shiny new package. PTAL. 

CI failure is a pre-existing flake. Filed https://github.com/tailscale/tailscale/issues/3247.

**Review by crawshaw — APPROVED** [review]

**Pull request merged**